libcudacxx/test/libcudacxx/std/containers/sequences/inplace_vector/assign.pass.cpp

```diff
@@ -0,0 +1,353 @@
+  }
+  {
+    using iter                            = cpp17_input_iterator<const T*>;
+    const cuda::std::array<T, 0> expected = {};
+    cuda::std::inplace_vector<T, 42> input_iter_iter_empty{};
+    input_iter_iter_empty.assign(iter{expected.begin()}, iter{expected.end()});
+    assert(input_iter_iter_empty.empty());
+  }
+
+  {
+    using iter                            = cpp17_input_iterator<const T*>;
+    const cuda::std::array<T, 0> expected = {};
+    cuda::std::inplace_vector<T, 42> input_iter_iter_shrink_empty{T(1), T(42), T(1337), T(0)};
+    input_iter_iter_shrink_empty.assign(iter{expected.begin()}, iter{expected.end()});
+    assert(input_iter_iter_shrink_empty.empty());
+  }
+
+  {
+    using iter                            = cpp17_input_iterator<const T*>;
+    const cuda::std::array<T, 2> expected = {T(42), T(42)};
+    cuda::std::inplace_vector<T, 42> input_iter_iter_shrink{T(1), T(42), T(1337), T(0)};
+    input_iter_iter_shrink.assign(iter{expected.begin()}, iter{expected.end()});
+    assert(!input_iter_iter_shrink.empty());
+    assert(
+      cuda::std::equal(input_iter_iter_shrink.begin(), input_iter_iter_shrink.end(), expected.begin(), expected.end()));
+  }
+
+  {
+    using iter                            = cpp17_input_iterator<const T*>;
+    const cuda::std::array<T, 6> expected = {T(42), T(42), T(42), T(42), T(42), T(42)};
+    cuda::std::inplace_vector<T, 42> input_iter_iter_grow{T(1), T(42), T(1337), T(0)};
+    input_iter_iter_grow.assign(iter{expected.begin()}, iter{expected.end()});
+    assert(!input_iter_iter_grow.empty());
+    assert(
+      cuda::std::equal(input_iter_iter_grow.begin(), input_iter_iter_grow.end(), expected.begin(), expected.end()));
+  }
+
+  {
+    const cuda::std::array<T, 0> expected = {};
+    cuda::std::inplace_vector<T, 42> iter_iter_empty{};
+    iter_iter_empty.assign(expected.begin(), expected.end());
+    assert(iter_iter_empty.empty());
+  }
+
+  {
+    const cuda::std::array<T, 0> expected = {};
+    cuda::std::inplace_vector<T, 42> iter_iter_shrink_empty{T(1), T(42), T(1337), T(0)};
+    iter_iter_shrink_empty.assign(expected.begin(), expected.end());
+    assert(iter_iter_shrink_empty.empty());
+  }
+
+  {
+    const cuda::std::array<T, 2> expected = {T(42), T(42)};
+    cuda::std::inplace_vector<T, 42> iter_iter_shrink{T(1), T(42), T(1337), T(0)};
+    iter_iter_shrink.assign(expected.begin(), expected.end());
+    assert(!iter_iter_shrink.empty());
+    assert(cuda::std::equal(iter_iter_shrink.begin(), iter_iter_shrink.end(), expected.begin(), expected.end()));
+  }
+
+  {
+    const cuda::std::array<T, 6> expected = {T(42), T(42), T(42), T(42), T(42), T(42)};
+    cuda::std::inplace_vector<T, 42> iter_iter_grow{T(1), T(42), T(1337), T(0)};
+    iter_iter_grow.assign(expected.begin(), expected.end());
+    assert(!iter_iter_grow.empty());
+    assert(cuda::std::equal(iter_iter_grow.begin(), iter_iter_grow.end(), expected.begin(), expected.end()));
```

Comment:
The name `iter_iter_grow` does not help me, but rather makes reading harder, because I have to find out how you called the vector in each test block.

Suggestion: maybe consistenly name the inplace vector something simple, maybe just `vec` or `v`. If the current names are meant to tell what is tested, maybe move that into a comment.

```suggestion
    // grow vector by assigning two iterators
    const cuda::std::array<T, 6> expected = {T(42), T(42), T(42), T(42), T(42), T(42)};
    cuda::std::inplace_vector<T, 42> vec{T(1), T(42), T(1337), T(0)};
    vec.assign(expected.begin(), expected.end());
    assert(!vec.empty());
    assert(cuda::std::equal(vec.begin(), vec.end(), expected.begin(), expected.end()));
```

Reply:
Yeah the intend is that the name of the vector describes the thing we test.

I am a bit loath to change it because I am used to all those old libc++ tests that had all kind of single letter names and it at times is really difficult to follow what is going on.

Mind that here individual test code is rather short, but I want to avoid having to scroll around to get what is actually tested

Reply:
Already changed almost all of it and it is much better now